backend/src/ml_space_lambda/utils/iam_manager.py

```diff
@@ -386,15 +386,20 @@ def _create_iam_policy(
         policy_identifier: str,
         policy_version: int,
     ) -> str:
+        tags = [
+            {"Key": policy_type.lower(), "Value": policy_identifier},
+            {"Key": "policyVersion", "Value": str(policy_version)},
+            {"Key": "system", "Value": self.system_tag},
+        ]
+
+        if policy_type.lower() == "user":
+            tags.append({"Key": "dynamic-user-role", "Value": "true"})
```

Comment:
This is a policy, not a role. I think you meant to add this to `_create_iam_role`

Reply:
You are absolutely right, great catch!